File: src/main/java/me/leodev/trollingfreedomreborn/trolls/explosion/KittyCannon.java
package me.leodev.trollingfreedomreborn.trolls.explosion;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.Sound;
import org.bukkit.entity.Entity;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Ocelot;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;

import java.util.ArrayList;

public class KittyCannon implements Listener {

    public static ArrayList<String> Kitty1 = new ArrayList();

    public void KittyCannon(final Player p) {
        String p2 = p.getName();


        Kitty1.add(p.getName());
        int repeat = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, new Runnable() {
            public void run() {
                EntityType cat = EntityType.OCELOT;
                Location twoBlocksAway = p.getEyeLocation().add(p.getEyeLocation().getDirection().multiply(10));
                final Ocelot ocelot = (Ocelot) p.getWorld().spawn(twoBlocksAway, cat.getEntityClass());
                ocelot.setVelocity(twoBlocksAway.getDirection().multiply(-2));
                p.getWorld().createExplosion(ocelot.getLocation(), 0F);
                p.playSound(ocelot.getLocation(), Sound.ENTITY_OCELOT_HURT, 100F, 100F);
                Location loc = ocelot.getLocation();
            }
        }, 10L, 10L);
        int delay = Bukkit.getScheduler().scheduleSyncDelayedTask(Core.instance, new Runnable() {
            public void run() {
                for (Entity entity : p.getNearbyEntities(100, 100, 100)) {
                    if (entity instanceof Ocelot)
                        entity.remove();
                }
            }
        }, 60L); // 600L (ticks) is equal to 30 seconds (20 ticks = 1 second)
        Core.instance.addTask(p, "kittycannon", repeat);
        Core.instance.addTask(p, "kittycannon", delay);
    }

    public void UnKittyCannon(Player p) {

        // String nick = user.getNick();
        if (Kitty1.contains(p.getName())) {
            for (Entity entity : p.getNearbyEntities(100, 100, 100)) {
                if (entity instanceof Ocelot)
                    entity.remove();
            }
            Kitty1.remove(p.getName());
        } else {
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/explosion/KittyCannon.kt
package com.leomadrassi.trollingfreedomreborn.trolls.explosion

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.Bukkit
import org.bukkit.Sound
import org.bukkit.entity.Entity
import org.bukkit.entity.Ocelot
import org.bukkit.entity.Player
import org.bukkit.event.Listener

class KittyCannon : Listener {

    companion object {
        val Kitty1 = ArrayList<String>()
    }

    fun KittyCannon(p: Player) {
        Kitty1.add(p.name)
                    val repeat = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, Runnable {
            val twoBlocksAway = p.eyeLocation.add(p.eyeLocation.direction.multiply(10))
            val ocelot = p.world.spawn(twoBlocksAway, Ocelot::class.java)
            ocelot.setVelocity(twoBlocksAway.direction.multiply(-2))
            p.world.createExplosion(ocelot.location, 0f)
            p.playSound(ocelot.location, Sound.ENTITY_OCELOT_HURT, 100f, 100f)
        }, 10L, 10L)
        val delay = Bukkit.getScheduler().scheduleSyncDelayedTask(Core.instance, Runnable {
            for (entity in p.getNearbyEntities(100.0, 100.0, 100.0)) {
                if (entity is Ocelot) {
                    entity.remove()
                }
            }
        }, 60L)
        Core.instance.addTask(p, "kittycannon", repeat)
        Core.instance.addTask(p, "kittycannon", delay)
    }

    fun UnKittyCannon(p: Player) {
        if (Kitty1.contains(p.name)) {
            for (entity in p.getNearbyEntities(100.0, 100.0, 100.0)) {
                if (entity is Ocelot) {
                    entity.remove()
                }
            }
            Kitty1.remove(p.name)
        }
    }
}
